src/test/java/org/jboss/xavier/integrations/EndToEndIT.java

```diff
@@ -0,0 +1,240 @@
+    }
+
+    private static Producer<String, String> createKafkaProducer() {
+        KafkaProducer<String, String> producer = new KafkaProducer<>(
+                ImmutableMap.of(
+                        ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, kafka.getBootstrapServers(),
+                        ProducerConfig.CLIENT_ID_CONFIG, UUID.randomUUID().toString()
+                ),
+                new StringSerializer(),
+                new StringSerializer()
+        );
+
+        return producer;
+    }
+        
+    @Test
+    public void end2endTest() throws Exception {
+        // given
+        camelContext.setTracing(true);
+        camelContext.start();
+
+        // when
+        // Start the camel route as if the UI was sending the file to the Camel Rest Upload route
+        new RestTemplate().postForEntity("http://localhost:8080/api/xavier/upload", getRequestEntityForRESTCall(), String.class);
+
+        // then
+        Thread.sleep(5000); //TODO check another approach
+
+        // Check database
+        Page<InitialSavingsEstimationReportModel> reports = reportService.findReports();
+        assertThat(Streams.stream(reports.iterator()).count()).isGreaterThan(0);
+        assertThat(Streams.stream(reports.iterator()).anyMatch(e -> e.getEnvironmentModel().getHypervisors() == 2)).isTrue();
+        assertThat(Streams.stream(reports.iterator()).anyMatch(e -> e.getSourceCostsModel().getYear1Server() == 42)).isTrue();
+
+        // Check the rest endpoint to retrieve the report
+        Map<String, String> params = new HashMap<>();
+        params.put("summary", "false");
+        String responseReport = new RestTemplate().getForObject("http://localhost:8080/api/xavier/report?summary={summary}", String.class, params);
+        assertThat(responseReport).contains("\"year1RhvGrandTotalValue\":187000.0");
+        assertThat(responseReport).contains("\"hypervisors\":2,\"year1Hypervisor\":20");
+        
+        camelContext.stop();
+    }
+
+    @NotNull
+    private HttpEntity<MultiValueMap<String, Object>> getRequestEntityForRESTCall() throws IOException {
+
+        // Headers
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.MULTIPART_FORM_DATA);
+        headers.set("x-rh-insights-request-id", "2544925e825b4f3f9418c88556541776");
+        headers.set("x-rh-identity", "eyJlbnRpdGxlbWVudHMiOnsiaW5zaWdodHMiOnsiaXNfZW50aXRsZWQiOnRydWV9LCJvcGVuc2hpZnQiOnsiaXNfZW50aXRsZWQiOnRydWV9LCJzbWFydF9tYW5hZ2VtZW50Ijp7ImlzX2VudGl0bGVkIjpmYWxzZX0sImh5YnJpZF9jbG91ZCI6eyJpc19lbnRpdGxlZCI6dHJ1ZX19LCJpZGVudGl0eSI6eyJpbnRlcm5hbCI6eyJhdXRoX3RpbWUiOjAsImF1dGhfdHlwZSI6Imp3dC1hdXRoIiwib3JnX2lkIjoiNjM0MDA1NiJ9LCJhY2NvdW50X251bWJlciI6IjE0NjAyOTAiLCJ1c2VyIjp7ImZpcnN0X25hbWUiOiJNYXJjbyIsImlzX2FjdGl2ZSI6dHJ1ZSwiaXNfaW50ZXJuYWwiOnRydWUsImxhc3RfbmFtZSI6IlJpenppIiwibG9jYWxlIjoiZW5fVVMiLCJpc19vcmdfYWRtaW4iOmZhbHNlLCJ1c2VybmFtZSI6Im1yaXp6aUByZWRoYXQuY29tIiwiZW1haWwiOiJtcml6emkrcWFAcmVkaGF0LmNvbSJ9LCJ0eXBlIjoiVXNlciJ9fQ==");
```

Comment:
Just a query -  is there a reason we're using different identity here to the one in src/test/resources/platform.upload.xavier-with-targz.json?

Reply:
Good catch.
Let me check why :+1:

Reply:
Made all the same to be consistent
Done